### PR TITLE
ResultTable: ability to specify default display function for cells

### PR DESCRIPTION
**Pull request by stellarhoof**
With this we can show highlighted fields without having to override their display functions.

### CODE REVIEW

**Comment by decrapifier** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>The README has not been updated. Please update the README.</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>We were able to automatically fix some formatting issues in this PR for you!</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>Could not find any browser results.</td>
    </tr>
  </tbody>
</table>


### Some things that were possibly fixed:

- Code that could be fixed via the --fix flag
- Formatting that could be fixed by prettier

Take a look at this commit to see what happened in detail: df434d8d4e402107f515391cd2beb8228f0cb9ef


And look at this wiki page to see the reasoning behind the ESLint rules: https://github.com/smartprocure/eslint-config-smartprocure/wiki/Rules-and-Why-We-Chose-Them
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against b8b70e915c8b1d716b04d1de5e6cdbfaacd15984
</p>

File: src/exampleTypes/ResultTable/TableBody.js [hunk 1]
```diff
@@ -1,13 +1,22 @@
...
 import { withTheme } from '../../utils/theme'
 import { StripedLoader } from '../../greyVest'
 
+let displayCell = (display, field, result) => {
+  let record = getRecord(result)
+  let data = _.get(field, record)
+  if (result.isBlank) return blankResult(display)(data, record)
+  if (display) return display(data, record)
+  if (_.isString(data))
+    return <span dangerouslySetInnerHTML={{ __html: data }} />
```
[the remark this answers is not in the document]
Comment (stellarhoof) [reply src/exampleTypes/ResultTable/TableBody.js:16]: Nice find! Will do.

[the remark this answers is not in the document]
Comment (stellarhoof) [reply src/exampleTypes/ResultTable/TableBody.js:16]: Unfortunately the sanitizer api is not well supported in browsers and the only polyfill is not an npm package. I'll use dompurify instead, which is used internally by the polyfill.

[the remark this answers is not in the document]
Comment (dshishkov) [reply src/exampleTypes/ResultTable/TableBody.js:16]: > Unfortunately the sanitizer api is not well supported in browsers and the only polyfill is not an npm package. I'll use dompurify instead, which is used internally by the polyfill.

Not sure if it is applicable here but, watch out for this gotcha with dompurify https://github.com/smartprocure/spark/blob/6ff4be9cc20097711afd0fdca3243288432c3d3f/web/src/util/format.js#L185

[the remark this answers is not in the document]
Comment (stellarhoof) [reply src/exampleTypes/ResultTable/TableBody.js]: We're safe here though because the results table is only used in the frontend.

**Review by rudolph9 — APPROVED** [review]

**Review by ccalbaugh — APPROVED** [review]

**Review by dshishkov — APPROVED** [review]

**Pull request merged**